### PR TITLE
Remove BZIP2 dependency

### PR DESCRIPTION
**Pull request by masterleinad**
While looking at the the `ZLIB` dependency I noticed that we are not using `BZIP2` anymore.
The less dependencies we have to maintain the better. :smiley:

### CODE REVIEW

**Comment by drwells** [comment]
Nice catch!

**Comment by drwells** [comment]
If you have time, would you please update `doc/users/cmake.html` and `doc/users/cmakelists.html` as part of this PR? That way we can get everything excised simultaneously.

**Comment by masterleinad** [comment]
@drwells Here you go!

File: doc/users/cmakelists.html [hunk 1]
```diff
@@ -842,21 +842,26 @@
...
 DEAL_II_WITH_METIS
 DEAL_II_WITH_MPI
 DEAL_II_WITH_MUPARSER
+DEAL_II_WITH_NANOFLANN
 DEAL_II_WITH_NETCDF
 DEAL_II_WITH_OPENCASCADE
 DEAL_II_WITH_P4EST
 DEAL_II_WITH_PETSC
 DEAL_II_WITH_SLEPC
+DEAL_II_WITH_SUNDIALS
```
Comment (drwells) [inline doc/users/cmakelists.html]: we have acquired a lot of stuff since the last time this was updated.

**Comment by drwells** [comment]
Let me ask a dumb question to make sure that I understand what is going on here.

We support (before this patch) using the bundled `boost::iostream` with both zlib and bzip2. This PR will just remove the option of using the bundled `boost::iostream` with bzip2, which we never use in the library. Is this correct?

**Comment by masterleinad** [comment]
> We support (before this patch) using the bundled boost::iostream with both zlib and bzip2. This PR will just remove the option of using the bundled boost::iostream with bzip2, which we never use in the library. Is this correct?

Yes, I am not even sure if this was used at any time in the last 4 years. At least `git log -SBZIP2 --since=2013.1.1 --until=2017.1.1 -- .` doesn't show anything suspicious.

**Comment by drwells** [comment]
I also can't find any evidence that we ever used this particular filter.

**Pull request merged**